### PR TITLE
Convert tests in dbt-adapter-tests to use new pytest framework

### LINKED ISSUE #4811: [CT-312] Convert dbt-adapters-tests
Converting the https://github.com/dbt-labs/dbt-adapter-tests tests to from yaml

### PR DESCRIPTION
**Pull request by gshank**
resolves #4811

### Description

This takes the tests in the dbt-adapter-tests repository and converts them to use the new pytest framework. Subsequent issues and pull requests will do the work to connect the tests up to the adapter repositories and enable running them on multiple adapters.

### Checklist

- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] I have updated the `CHANGELOG.md` and added information about my change

### CODE REVIEW

File: core/dbt/tests/fixtures/project.py [hunk 1]
```diff
@@ -310,38 +292,7 @@ def run_sql_file(self, sql_path):
 
     # run sql from a string, using adapter saved at test startup
     def run_sql(self, sql, fetch=None):
```
Comment (emmyoop) [inline core/dbt/tests/fixtures/project.py:294]: Can you rename this to differentiate it from the utils `run_sql`?

File: tests/functional/adapter/files.py [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+"""
+
+base_materialized_var_sql = materialized_var + model_base
+base_table_sql = materialized_table + model_base
+base_view_sql = materialized_view + model_base
+base_ephemeral_sql = materialized_ephemeral + model_base
+ephemeral_with_cte_sql = materialized_ephemeral + model_ephemeral_with_cte
+ephemeral_view_sql = materialized_view + model_ephemeral
+ephemeral_table_sql = materialized_table + model_ephemeral
+incremental_sql = materialized_incremental + model_incremental
```
Comment (emmyoop) [inline tests/functional/adapter/files.py:208]: nit: Can the config vars be prefixed with `config`?  It took me a minute to see what was being combined here.  Naming with `config` helps point that out.  Plus if we do something like this in other tests where config varies (but not just by materializations) we can name the same way.

File: tests/functional/adapter/test_base.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        "table_model": "table",
+        "swappable": "view",
+    }
+    check_relation_types(project.adapter, expected)
+
+    # run_dbt changing materialized_var to incremental
+    results = run_dbt(["run", "-m", "swappable", "--vars", "materialized_var: incremental"])
+    assert len(results) == 1
+
+    # check relation types, swappable is table
```
Comment (emmyoop) [inline tests/functional/adapter/test_base.py:92]: All the comments are so so so nice.

**Review by emmyoop — COMMENTED** [review]
Overall looks good to me.  Just a few names I'm concerned about.

File: core/dbt/tests/fixtures/project.py (hunk 1 above)
Comment (gshank) [reply core/dbt/tests/fixtures/project.py:294]: I renamed the other one, since is the one that we're actually using in the code.

File: tests/functional/adapter/files.py (hunk 2 above)
Comment (gshank) [reply tests/functional/adapter/files.py]: Yeah, good idea. These files were pretty confusing and adding config_ does help.

File: core/dbt/tests/adapter.py [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+def update_rows(adapter, update_rows_config):
+    """
+    {
+      "name": "base",
+      "dst_col": "some_date"
+      "clause": {
+          "type": "add_timestamp",
+          "src_col": "some_date",
+       "where" "id > 10"
+    }
```
Comment (ChenyuLInx) [inline core/dbt/tests/adapter.py:147]: This example is really helpful! I feel like we might want to do this in more places like adding the doc string. For me a lot of these things are new so it was harder to follow the function without an example arg.

File: core/dbt/tests/util.py [hunk 5]
```diff
@@ -85,3 +90,59 @@ def read_file(*paths):
...
+# For updating yaml config files
+def update_config_file(updates, *paths):
+    current_yaml = read_file(*paths)
+    config = yaml.safe_load(current_yaml)
+    config.update(updates)
+    new_yaml = yaml.safe_dump(config)
+    write_file(new_yaml, *paths)
+
+
+def run_sql(sql, adapter, database, schema, fetch=None):
```
Comment (ChenyuLInx) [inline core/dbt/tests/util.py]: how is this related to run_sql at [tables.py](https://github.com/dbt-labs/dbt-core/blob/8fd8dfcf74657ccc5314640b22cb2589eed8a824/core/dbt/tests/tables.py#L311)

File: tests/functional/adapter/files.py [hunk 6]
```diff
@@ -0,0 +1,208 @@
+seeds_base_csv = """
```
Comment (ChenyuLInx) [inline tests/functional/adapter/files.py:1]: I really like this, I will add this to the auto conversion script

**Review by ChenyuLInx — COMMENTED** [review]
Over all looks great! Maybe we want to start adding doc strings to function as a standard

File: core/dbt/tests/util.py (hunk 5 above)
Comment (gshank) [reply core/dbt/tests/util.py]: This one has been renamed to run_sql_with_adapter now. It does pretty much the same thing as in tables.py, but the plan is to get rid of tables.py when we can, so I'd rather leave refactoring it to that effort.

File: core/dbt/tests/adapter.py (hunk 4 above)
Comment (gshank) [reply core/dbt/tests/adapter.py:154]: I added a few more doc strings.

**Review by emmyoop — APPROVED** [review]
LGTM
